### ISSUE TITLE
--curses option for synapse_port_db is broken in python3

### ISSUE THREAD

**Issue by Half-Shot** [matrix-org/synapse#4596]
```
root@helios:~/synapse# synapse_port_db --curses --sqlite-database homeserver.db.snapshot --postgres-config homeserver-postgres.yaml
Traceback (most recent call last):
File "/usr/bin/synapse_port_db", line 542, in run
[self.handle_table(*res) for res in setup_res], consumeErrors=True
twisted.internet.defer.FirstError: FirstError[#34, [Failure instance: Traceback: <class 'AttributeError'>: 'dict_items' object has no attribute 'sort'
/usr/bin/synapse_port_db:542:run
/usr/bin/synapse_port_db:542:<listcomp>
/opt/venvs/matrix-synapse/lib/python3.6/site-packages/twisted/internet/defer.py:1613:unwindGenerator
/opt/venvs/matrix-synapse/lib/python3.6/site-packages/twisted/internet/defer.py:1529:_cancellableInlineCallbacks
--- <exception caught here> ---
/opt/venvs/matrix-synapse/lib/python3.6/site-packages/twisted/internet/defer.py:1418:_inlineCallbacks
/usr/bin/synapse_port_db:281:handle_table
/usr/bin/synapse_port_db:771:update
/usr/bin/synapse_port_db:813:render
]]
```

Which is a fail in handle_table trying to `sort()` some tables. I presume this is an artifact of the py3 port?

**Comment by AkselRocks** [comment]
Same problem here.
```
sudo synapse_port_db -v --curses --sqlite-database /var/lib/matrix-synapse/homeserver.db.snapshot --postgres-config /etc/matrix-synapse/homeserver-postgres.yaml
[sudo] Passwort für aadm:
Traceback (most recent call last):
  File "/usr/bin/synapse_port_db", line 542, in run
    [self.handle_table(*res) for res in setup_res], consumeErrors=True
twisted.internet.defer.FirstError: FirstError[#63, [Failure instance: Traceback: <class 'AttributeError'>: 'dict_items' object has no attribute 'sort'
/usr/bin/synapse_port_db:542:run
/usr/bin/synapse_port_db:542:<listcomp>
/opt/venvs/matrix-synapse/lib/python3.5/site-packages/twisted/internet/defer.py:1613:unwindGenerator
/opt/venvs/matrix-synapse/lib/python3.5/site-packages/twisted/internet/defer.py:1529:_cancellableInlineCallbacks
--- <exception caught here> ---
/opt/venvs/matrix-synapse/lib/python3.5/site-packages/twisted/internet/defer.py:1418:_inlineCallbacks
/usr/bin/synapse_port_db:281:handle_table
/usr/bin/synapse_port_db:771:update
/usr/bin/synapse_port_db:813:render
]]```

**Comment by avignat** [comment]
```
Traceback (most recent call last):
  File "/usr/bin/synapse_port_db", line 542, in run
    [self.handle_table(*res) for res in setup_res], consumeErrors=True
twisted.internet.defer.FirstError: FirstError[#34, [Failure instance: Traceback: <class 'AttributeError'>: 'dict_items' object has no attribute 'sort'
/usr/bin/synapse_port_db:542:run
/usr/bin/synapse_port_db:542:<listcomp>
/opt/venvs/matrix-synapse/lib/python3.5/site-packages/twisted/internet/defer.py:1613:unwindGenerator
/opt/venvs/matrix-synapse/lib/python3.5/site-packages/twisted/internet/defer.py:1529:_cancellableInlineCallbacks
--- <exception caught here> ---
/opt/venvs/matrix-synapse/lib/python3.5/site-packages/twisted/internet/defer.py:1418:_inlineCallbacks
/usr/bin/synapse_port_db:281:handle_table
/usr/bin/synapse_port_db:771:update
/usr/bin/synapse_port_db:813:render
]]

```

**Comment by ajensenwaud** [comment]
Ran into this error as well with Python 3 on Debian buster/sid.
Fixed with PR #4978

**Comment by richvdh** [comment]
fixed by #5004
